jaworski_cellseg/configuration_widget.py

```diff
@@ -0,0 +1,81 @@
+
+from magicgui import magicgui
+from typing import Optional
+import yaml
+
+def create_configuration_widget(configurations: list[str],
+                               current_config,
+                               config_file_path,
+                               jaworski_obj):
+    """ 
+     Widget to save and load values from the GUI
+    """
+    @magicgui(
+        call_button="Save Configuration",
+        auto_call=False,
+        configuration_names={
+            "label": "Config",
+            "widget_type": "ComboBox",
+            "choices": configurations,
+        },
+        save_confg={
+            "label": "Save",
+        },
+    )
+    def configuration_widget(configuration_names: str = configurations[current_config],
+                             save_confg: Optional[str] = ""):
+        
+        if configuration_widget.save_confg.value:
+            with open(config_file_path, "w") as file:
+                data = {}
+                data["gausianFilter"] = jaworski_obj.pre_process_data_widget.gaussian_checkbox.value
+                data["gaussian_sigma"] = jaworski_obj.pre_process_data_widget.gaussian_factor.value
+                data["contrast_adjustment"] = jaworski_obj.pre_process_data_widget.contrast_adjustment.value
+                data["contrast_min"] = jaworski_obj.pre_process_data_widget.contrast_lower.value
+                data["contrast_max"] = jaworski_obj.pre_process_data_widget.contrast_upper.value
+                data["binary_map_threshold"] =  jaworski_obj.pre_process_data_widget.binary_map_threshold.value
+                data["inference_use_window_choice"] = jaworski_obj.inferer_widget.use_window_choice.isChecked()
+                data["inference_window_size"] = int(jaworski_obj.inferer_widget.window_size_choice.currentText())
+                data["inference_input"] = jaworski_obj.inferer_widget.model_input_size.value()
+                data["inference_overlap"] = jaworski_obj.inferer_widget.window_overlap_slider.value()
+                data["inference_perform_threhold"] = jaworski_obj.inferer_widget.thresholding_checkbox.isChecked()
+                data["inference_perform_threhold_value"] = jaworski_obj.inferer_widget.thresholding_slider.value()
+                data["inference_instance_segmentation"] = jaworski_obj.inferer_widget.use_instance_choice.isChecked()
+                data["inference_instance_segmentation_option"] = "Voronoi-Otsu"
+                voronoi_widget = jaworski_obj.inferer_widget.instance_widgets.methods[ "Voronoi-Otsu" ]
+                data["inference_instance_segmentation_spot_signma"] = voronoi_widget.counters[0].value()
+                data["inference_instance_segmentation_outline_signma"] = voronoi_widget.counters[1].value()
+                data["inference_instance_segmentation_small_object_removal"] = voronoi_widget.counters[2].value()
```

Comment:
Consider a helper function to build the dict with the widget data:
```python
def extract_widget_values(jaworski_obj):
    pre_process = jaworski_obj.pre_process_data_widget
    inferer = jaworski_obj.inferer_widget
    voronoi_widget = inferer.instance_widgets.methods["Voronoi-Otsu"]
    
    return {
        "gausianFilter": pre_process.gaussian_checkbox.value,
        "gaussian_sigma": pre_process.gaussian_factor.value,
        "contrast_adjustment": pre_process.contrast_adjustment.value,
        "contrast_min": pre_process.contrast_lower.value,
        "contrast_max": pre_process.contrast_upper.value,
        "binary_map_threshold": pre_process.binary_map_threshold.value,
        "inference_use_window_choice": inferer.use_window_choice.isChecked(),
        "inference_window_size": int(inferer.window_size_choice.currentText()),
        "inference_input": inferer.model_input_size.value(),
        "inference_overlap": inferer.window_overlap_slider.value(),
        "inference_perform_threhold": inferer.thresholding_checkbox.isChecked(),
        "inference_perform_threhold_value": inferer.thresholding_slider.value(),
        "inference_instance_segmentation": inferer.use_instance_choice.isChecked(),
        "inference_instance_segmentation_option": "Voronoi-Otsu",
        "inference_instance_segmentation_spot_signma": voronoi_widget.counters[0].value(),
        "inference_instance_segmentation_outline_signma": voronoi_widget.counters[1].value(),
        "inference_instance_segmentation_small_object_removal": voronoi_widget.counters[2].value()
    }

Reply:
I moved all the initialization to a helper function